### ISSUE TITLE
handle complex normalization with clipping

### ISSUE THREAD

**Issue by rbavery** [crim-ca/dlm-extension#8]
SATLAS, a project that produces pretrained models that generate featuremaps, finetuned task specific models, and a dataset, provides normalization steps that #2 doesn't capture: https://github.com/allenai/satlas/blob/main/Normalization.md#sentinel-2-images Different steps are applied for different bands.

> TCI (this is three channels, RGB; divide by 255)
> B05, B06, B07, B08, B11, B12 (divide by 8160, clip to 0-1)

I think b05-b12 have the extra clipping step because of more sever outliers.

The normalization step is like max norm per band, but instead of max for bands B05-B12, they use a different arbitrary value and then clip to the expected 0-1 range after division. While max norm would produce an equivalent range, the input distribution would be skewed.

So I think we need to represent this kind of normalization. Maybe we can make a new norm_type called "clip_norm" and specify "divide_value"= 8160 and "clip_range"=(0,1). 

also, this type of normalization is only applied to some bands. so maybe we need to specify norm_type per band in a list rather than as a scalar for the entire model input.

**Comment by fmariv** [comment]
Right now, and as I see in the [PR](https://github.com/crim-ca/dlm-extension/blob/f9f66d633e574c4c1a0b4e790ec5367fda583019/README.md#model-input-object), this has been solved as `norm_with_clip` and then indicate the values ​​by which to make the clip for each band in `norm_with_clip_values`. Looks good to me, as it allows to use different values ​​for different bands and reading the description in the README it is well understood. Important is the part about _each value is used to divide each band before clipping values ​​between 0 and 1._, otherwise could be confusing.

**Comment by fmigneault** [comment]
Fixed by #2
